# Coordinate testing

In [1]:
from __future__ import print_function, division
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

from astropy.io import fits

In [2]:
# Test set of binary matches from APW
df1 = pd.DataFrame(fits.getdata('../data/tim-binaries-1.fits'))
df2 = pd.DataFrame(fits.getdata('../data/tim-binaries-2.fits'))

In [10]:
import os
from astroquery.vizier import Vizier
from astropy.coordinates import SkyCoord
import astropy.units as u

dt = 15*u.yr # back to c. 2000
i = 59 # just an example
ra = df1.ra[i]*u.deg - dt*df1.pmra[i]*u.mas/u.yr
dec = df1.dec[i]*u.deg - dt*df1.pmdec[i]*u.mas/u.yr
q = Vizier.query_region(SkyCoord(ra, dec), radius=5*u.arcsec,
                                catalog=['2mass'])
q[0]

<Table masked=True length=2>
 _RAJ2000   _DEJ2000     _r    RAJ2000   ... scanKey coaddKey coadd Opt 
   deg        deg       arcs     deg     ...                            
 float64    float64   float64  float64   ...  int32   int32   int16 str3
---------- ---------- ------- ---------- ... ------- -------- ----- ----
 24.692811  63.717216   4.347  24.692811 ...   41900   963686   103  Opt
 24.694572  63.716732   1.154  24.694572 ...   41900   963686   103  Opt

The closest 2mass source is still 1.15 arcsec away from where this star should be, given the current coords and the proper motion.  What's even stranger is that the Tycho-2 query doesn't seem to match up either:

In [11]:
dt = (2015-1991.5)*u.yr # back to c. 1991.5 (what the Tycho-2 website says was avg. observation epoch)
i = 59
ra = df1.ra[i]*u.deg - dt*df1.pmra[i]*u.mas/u.yr
dec = df1.dec[i]*u.deg - dt*df1.pmdec[i]*u.mas/u.yr
q = Vizier.query_region(SkyCoord(ra, dec), radius=5*u.arcsec,
                                catalog=['tycho2'])
q[0]

<Table masked=True length=1>
 _RAJ2000   _DEJ2000     _r    TYC1  TYC2 ... e_RAdeg e_DEdeg posflg   corr 
   deg        deg       arcs              ...   mas     mas                 
 float64    float64   float64 int16 int16 ... float32 float32  str1  float32
---------- ---------- ------- ----- ----- ... ------- ------- ------ -------
 24.694258  63.716646   1.291  4035   103 ...    17.7    12.4           -0.1

Still 1.2 arcsec away?  I must be misinterpreting the meaning of either the TGAS coordinates or the proper motions.  Or using astropy incorrectly?